code/modules/vending/mail.dm

```diff
@@ -0,0 +1,308 @@
+			if (main_department == sorting_dept)
+				sorted_mail.Add(some_mail)
+				sorted ++
+		else
+			unable_to_sort ++
+	if (length(sorted_mail) == 0)
+		currentstate = STATE_NO
+		update_appearance()
+		playsound(src, 'sound/machines/buzz/buzz-sigh.ogg', 20, TRUE)
+		say("No mail for the following department: [sorting_dept].")
+	else
+		currentstate = STATE_YES
+		update_appearance()
+		say("[sorted] envelope\s sorted successfully.")
+		playsound(src, 'sound/machines/ping.ogg', 20, TRUE)
+		to_chat(usr, span_notice("[src] ejects [length(sorted_mail)] envelope\s."))
+		var/turf/unload_turf = get_unload_turf()
+		for (var/obj/item/mail/mail_in_list in sorted_mail)
+			mail_in_list.forceMove(unload_turf)
+			sorted_mail -= mail_in_list
+			mail_list -= mail_in_list
+	sleep(10)
+	if (unable_to_sort > 0)
+		playsound(src, 'sound/machines/buzz/buzz-sigh.ogg', 20, TRUE)
+		say("Couldn't sort [unable_to_sort] envelope\s.")
+	else
+		playsound(src, 'sound/machines/ping.ogg', 20, TRUE)
+		say("[total_to_sort] envelope\s processed.")
+	sleep(10)
+	currentstate = STATE_IDLE
+	update_appearance()
+
+/obj/machinery/mailsorter/attackby(obj/item/thingy, mob/user, params)
+	if (istype(thingy, /obj/item/storage/bag/mail))
+		if (length(thingy.contents) < 1)
+			to_chat(user, span_warning("The [thingy] is empty!"))
+			return
+		var/loaded = 0
+		for (var/obj/item/mail in thingy.contents)
+			if (!(mail.item_flags & ABSTRACT) && \
+				!(mail.flags_1 & HOLOGRAM_1) && \
+				accept_check(mail) \
+			)
+				if (length(mail_list) + 1 > mail_capacity)
+					to_chat(user, span_warning("There is no space for more mail in [src]!"))
+					return FALSE
+				else if (load(mail, usr))
+					loaded++
+					mail_list += mail
+		if(loaded)
+			user.visible_message(span_notice("[user] loads \the [src] with \the [thingy]."), \
+			span_notice("You load \the [src] with \the [thingy]."))
+			if(length(thingy.contents))
+				to_chat(user, span_warning("Some items are refused."))
+			return TRUE
+		else
+			to_chat(user, span_warning("There is nothing in \the [thingy] to put in the [src]!"))
+			return FALSE
+	else if (istype(thingy, /obj/item/mail))
+		if (length(mail_list) + 1 > mail_capacity)
+			to_chat(user, span_warning("There is no space for more mail in [src]!"))
+		else
+			thingy.forceMove(src)
+			mail_list += thingy
+			to_chat(user, span_notice("The [src] whizzles as it accepts the [thingy]."))
+	. = ..()
+
+/obj/machinery/mailsorter/proc/pick_mail(usr)
+	if(!length(mail_list))
+		return
+	var/obj/item/mail/mail_throw = tgui_input_list(usr, "Choose the envelope to eject","Mail Sorting", mail_list)
+	if(!mail_throw)
+		return
+	currentstate = STATE_SORTING
+	update_appearance()
+	if (!sort_delay())
+		return
+	to_chat(usr, span_notice("[src] reluctantly spits out [mail_throw]."))
+	var/turf/unload_turf = get_unload_turf()
+	mail_throw.forceMove(unload_turf)
+	mail_throw.throw_at(unload_turf, 2, 3)
+	mail_list -= mail_throw
+	currentstate = STATE_IDLE
+	update_appearance()
```

Comment:
updating icon? ``UPDATE_ICON``. Only overlay/icon state? ``UPDATE_OVERLAYS`` / ``UPDATE_ICON_STATE`` arg

Reply:
What's so wrong about updating all of them? I'll do that, but just why...

Reply:
It's better to be specific so people reading know why we're updating appearances